### PR TITLE
Amplify/broaden def of 'verification method'

### PR DESCRIPTION
**Pull request by dhh1128**
Signed-off-by: Daniel Hardman <daniel.hardman@gmail.com>

See https://github.com/w3c/did-core/issues/276

### CODE REVIEW

**Comment by peacekeeper** [comment]
The only aspect of this PR I find problematic is that it adds a lot of text to the Terminology section. All other terms in that section consist of a single paragraph, whereas this one has three. Perhaps most of this new text should be moved elsewhere (especially if a new `verificationMethod` core property is introduced - as suggested in https://github.com/w3c/did-core/issues/283).

File: terms.html [hunk 1]
```diff
@@ -261,9 +261,26 @@
...
-proof.
+    <p>
+A set of parameters that can be used to independently verify a proof according to
+a particular method. For example, a public key can be used as a verification
+method with respect to a digital signature; in such usage, it verifies that the
+signer possessed the associated private key.
+    </p>
+    <p>
+"Verification" and "proof" in this definition are intended to apply broadly. A public
+key may be used during Diffie-Hellman key exchange to negotiate a shared symmetric
```
Comment (msporny) [inline terms.html:272]:
```suggestion
key might be used during Diffie-Hellman key exchange to negotiate a shared symmetric
```

File: terms.html [hunk 2]
```diff
@@ -261,9 +261,26 @@
...
+key for encryption. This guarantees the integrity of the key agreement process. It
+is thus another type of verification method, even though descriptions of the process
+may not use the words "verification" and "proof."
```
Comment (msporny) [inline terms.html:275]:
```suggestion
might not use the words "verification" and "proof."
```

File: terms.html [hunk 3]
```diff
@@ -261,9 +261,26 @@
...
+    </p>
+    <p>
+Verification methods may take many parameters. An example of this might be a set of five
```
Comment (msporny) [inline terms.html:278]:
```suggestion
Verification methods might take many parameters. An example of this is a set of five
```

File: terms.html [hunk 4]
```diff
@@ -261,9 +261,26 @@
...
+cryptographic keys from which any three must contribute to a threshold signature. Methods
```
Comment (msporny) [inline terms.html:279]:
```suggestion
cryptographic keys from which any three are required to contribute to a threshold signature. Methods
```

**Pull request merged**